core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/Purchases.ios.kt

```diff
@@ -264,8 +268,17 @@ public actual class Purchases private constructor(private val iosPurchases: IosP
             completion = { storeTransactionMap, error ->
                 if (error != null) onError(error.error().toPurchasesErrorOrThrow())
 
-                // TODO: Parse storeTransactionMap to a StoreTransaction object
-                else onSuccess()
+                // FIXME: Handle this case better. Maybe use onError()?
+                if (storeTransactionMap == null) error("Expected a non-null RCStoreTransactionMap")
```

Comment:
Is a null map a valid state? I'd guess not. 

If it is, we might want to call `onError()` with an appropriate error message.